src/es/upc/fib/prop/usParlament/driver/GeneralDriver.java

```diff
@@ -58,50 +56,59 @@ public static void mpManagement() {
         Boolean seguir = true;
         while(seguir) {
             System.out.println("Node Management:");
-            System.out.println("1-Enter MP's\n2-Erase MP\n3-Get an MP with its attributes\n4-Get all MP's\n5-Get Common Attributes\nany other key-EXIT");
+            System.out.println("1-Enter MP's\n2-Erase MP\n3-Get an MP with its attributes\n4-Get all MP's\n5-Get Common Attributes\nany other number-EXIT");
             Integer num = read.nextInt();
-            read.nextLine();
+            readCommand(num);
             String fullname;
             String st;
             State estat;
             Integer dist;
+            Boolean readmp = true;
             MP p1;
             switch(num) {
                 case 1:
-                    System.out.println("Enter the full name, the state and the district of the MP's\nafter entering anything hit enter\nwhen you're done instead of the name");
+                    System.out.println("Enter the full name, the state and the district of the MP's\nafter entering anything hit enter\nwhen you're done instead of the name enter 0");
                     while(true) {
                         System.out.print("fullname: ");
                         fullname = read.nextLine();
                         if (fullname.equals("0")) {System.out.println('\n'); break;}
                         System.out.print("state: ");
-                        st = read.next();
-                        estat = State.valueOf(st);
-                        System.out.print("district :");
-                        dist = read.nextInt();
-                        read.nextLine();
-                        MP p = new MP(fullname, estat, dist);
-                        c.addNode(p);
-                        System.out.println("MP added successfully");
+                        try {
+                            st = read.nextLine();
+                            estat = State.valueOf(st);
+                            System.out.print("district :");
+                            dist = read.nextInt();
+                            MP p = new MP(fullname, estat, dist);
+                            if (c.getMP(estat, dist) == null) c.addNode(p);
+                            else System.out.println("There's an MP already assigned to this state and district");
+                            System.out.println("MP added successfully");
+                            break;
+                        } catch (IllegalArgumentException i) {
```

Comment:
I think IllegalArgumentException catch only 'estat = State.valueOf(st); But if you put string instead of int as input for dist = read.nextIn() It will throw InputMismatchException and program will crash. So you should catch also this exception. (Im not absolutely sure try it. :-) )